src/main/scala/firrtl/Emitter.scala

```diff
@@ -325,6 +325,7 @@ class VerilogEmitter extends SeqTransform with Emitter {
      def c1: Int = doprim.consts(1).toInt
 
      def checkArgumentLegality(e: Expression): Unit = e match {
+       case _ if e.tpe == Utils.BoolType =>
```

Comment:
Parens would make the emitter more robust -- while the checks in the inlining pass are robust, this adds a blanket ability for the emitter to accept expression trees and emit them in a manner that might not reflect Verilog precedence.

Reply:
Agreed @albert-magyar, this seems bad to expose, although I'm not sure a better way to do it. Any ideas @albertchen-sifive  or @albert-magyar ?

Reply:
I can add the same precedence checks that are in `InlineBooleanExpressions` here.